lib/xboxrt/vcruntime/excpt.cpp

```diff
@@ -0,0 +1,201 @@
+#include <excpt.h>
+#include <excpt_cxx.hpp>
+#include <eh.h>
+#include <xboxkrnl/xboxkrnl.h>
+
+#define MAGIC_VC  0x19930520 // up to VC6
+#define MAGIC_VC7 0x19930521 // VC7.x(2002-2003)
+#define MAGIC_VC8 0x19930522 // VC8 (2005)
+
+__declspec(noreturn) extern "C" void __stdcall _CxxThrowException (void *pExceptionObject, _ThrowInfo *pThrowInfo)
+{
+    DbgPrint("Throwing object at %x\n", pExceptionObject);
+    DbgPrint("throw info at %x\n", pThrowInfo);
+
+    EXCEPTION_RECORD exceptionRecord;
+    exceptionRecord.ExceptionCode = 0xE06D7363; // FIXME: Magic number for C++ exceptions
+    exceptionRecord.ExceptionRecord = nullptr;
+    exceptionRecord.ExceptionAddress = (PVOID)_CxxThrowException;
+    exceptionRecord.ExceptionFlags = EXCEPTION_NONCONTINUABLE;
+    exceptionRecord.NumberParameters = 3;
+    exceptionRecord.ExceptionInformation[0] = MAGIC_VC8;
```

Comment:
Would we ever use `MAGIC_VC` or `MAGIC_VC7` here?
Why was this version chosen?

Reply:
VC8 exception handling ABI is what clang/LLVM use currently. We shouldn't have to ever use a different version here.

Reply:
Why do we need the other values then? We don't seem to use them in this file?
If you want to add support for others (maybe for catching exceptions from a precompiled lib, or forcing an old version?) we should already prepare the affected switch/cases maybe?